docs/proposals/configuration.md

```diff
@@ -0,0 +1,740 @@
+	ResponseModule string `json:"responseModule,omitempty"`
+}
+
+type KubeConfig struct {
+	// Master set the address of the Kubernetes API server (overrides any value in Kubeconfig)
+	// such as https://127.0.0.1:8443
+	// default ""
+	Master string `json:"master,omitempty"`
+	// Kubeconfig set path to kubeconfig file with authorization and master location information.
+	// default "/root/.kube/config"
+	Kubeconfig string `json:"kubeconfig,omitempty"`
+}
+
+```
+
+#### edgecore config apis
+
+defined in `pkg/apis/edgecore/v1alpha1/types.go` dir
+
+```go
+
+import metaconfig "github.com/kubeedge/kubeedge/pkg/apis/meta/v1alpha1"
+
+type EdgeCoreConfig struct {
+	metaconfig.TypeMeta
+	// Mqtt set mqtt config for edgecore
+	// +Required
+	Mqtt MqttConfig `json:"mqtt,omitempty"`
+	// EdgeHub set edgehub module config
+	// +Required
+	EdgeHub EdgeHubConfig `json:"edgehub,omitempty"`
+	// Edged set edged module config
+	// +Required
+	Edged EdgedConfig `json:"edged,omitempty"`
+	// Mesh set mesh module config
+	// +Required
+	Mesh MeshConfig `json:"mesh,omitempty"`
+	// Modules set which modules are enabled
+	// +Required
+	Modules metaconfig.Modules `json:"modules,omitempty"`
+	// Metamanager set meta module config
+	// +Required
+	Metamanager Metamanager `json:"metamanager,omitempty"`
+}
+
+type MqttConfig struct {
+	// Server set external mqtt broker url
+	// default tcp://127.0.0.1:1883
+	Server string `json:"server,omitempty"`
+	// InternalServer set internal mqtt broker url
+	// default tcp://127.0.0.1:1884
+	InternalServer string `json:"internalServer,omitempty"`
+	// Mode set which broker type will be choose
+	// 0: internal mqtt broker enable only. 1: internal and external mqtt broker enable. 2: external mqtt broker enable only
+	// default: 0
+	Mode uint8 `json:"mode,omitempty"`
+	// QOS set mqtt qos
+	// 0: QOSAtMostOnce, 1: QOSAtLeastOnce, 2: QOSExactlyOnce
+	// default 0
+	QOS uint8 `json:"qos,omitempty"`
+	// Retain set whether server will store the message and can be delivered to future subscribers
+	// if this flag set true, sever will store the message and can be delivered to future subscribers
+	// default false
+	Retain bool `json:"retain,omitempty"`
+	// SessionQueueSize set size of how many sessions will be handled.
+	// default 100
+	SessionQueueSize int32 `json:"sessionQueueSize,omitempty"`
+}
+
+type EdgeHubConfig struct {
+	// WebSocket set websocket config for edgehub module
+	WebSocket WebSocketConfig `json:"webSocket,omitempty"`
+	// Quic set quic config for edgehub module
+	Quic QuicConfig `json:"quic,omitempty"`
+	// Controller set controller config for edgehub module
+	Controller ControllerConfig `json:"controller,omitempty"`
+}
+
+type WebSocketConfig struct {
+	// Server set websocket server address (ip:port)
+	Server string `json:"server,omitempty"`
+	// TLSCertFile is the file containing x509 Certificate for HTTPS
+	// default /etc/kubeedge/certs/edge.crt
+	TLSCertFile string `json:"tlsCertFile,omitempty"`
```

Comment:
Both `WebSocketConfig` and `QuicConfig` have `TLSCertFile`, `TLSPrivateKeyFile`, we shall make them (including `TLSCAFile`) as EdgeHubConfig fields.

Reply:
ok